projects/cashmere/src/lib/input/maxlength.directive.ts

```diff
@@ -0,0 +1,36 @@
+import {AfterViewInit, Directive, ElementRef, HostListener,
+    Input, OnInit, Renderer2} from '@angular/core';
+
+    @Directive({
+        selector: '[hcMaxLength]'
+    })
+    export class MaxLengthDirective implements OnInit, AfterViewInit {
+        @Input() hcMaxLength: number;
+        private div: HTMLDivElement;
+
+        constructor(private el: ElementRef, private renderer: Renderer2) {}
+
+        @HostListener('input', ['$event']) onChange(event) {
+            this.update(event.target.value.length);
+        }
+
+        ngOnInit() {
+        this.renderer.setAttribute(this.el.nativeElement, 'maxLength', this.hcMaxLength.toString());
+        }
+
+        ngAfterViewInit() {
+            this.div = this.renderer.createElement('span');
+            this.div.classList.add('char-count');
+            this.renderer.insertBefore(this.el.nativeElement.parentNode.parentNode.parentNode, this.div, this.el.nativeElement.nextSibling);
+            this.update(this.el.nativeElement.value.length);
+        }
+
+        private update(length: number) {
+            if (length === this.hcMaxLength){
```

Comment:
Is it possible for the length to be greater than? Should it be `>=` instead?

Reply:
It's not possible because once the user hits the maxlength, typing in the box is disabled.